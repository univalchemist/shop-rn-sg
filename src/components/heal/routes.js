/* istanbul ignore file */

//appointment
export const HEAL_APPOINTMENT_LIST = 'HealAppointmentList';
export const HEAL_APPOINTMENT_DETAIL = 'HealAppointmentDetail';
export const HEAL_APPOINTMENT_CONFIRMATION = 'HealAppointmentConfirmation';
export const HEAL_THANK_YOU = 'HealThankYou';
export const HEAL_APPOINTMENT_REQUESTED = 'HealAppointmentRequested';

//medication
export const HEAL_MEDICATION = 'HealMedication';

//invoice
export const HEAL_INVOICES = 'HealInvoices';
export const HEAL_INVOICE_DETAILS = 'HealInvoiceDetails';

//doctor
export const HEAL_DOCTOR_LISTING = 'HealDoctorListing';
export const HEAL_SPECIALITY_LISTING = 'HealSpecialityListing';
export const HEAL_SCAN_QR_CODE = 'HealScanQrCode';
export const HEAL_DOCTOR_DETAIL = 'HealDoctorDetail';
export const HEAL_UNIFY_SEARCH = 'HealUnifySearch';
export const HEAL_MULTIPLE_CLINICS_LIST_VIEW = 'HealMultipleClinicsListView';
export const HEAL_CLINIC_DETAILS = 'HealClinicDetails';
export const HEAL_PANEL_CLINIC_DETAILS = 'HealPanelClinicDetails';
export const HEAL_SELECT_FAMILY_MEMBER = 'HealSelectFamilyMember';
export const HEAL_SELECT_CLINIC = 'HealSelectClinic';
export const HEAL_CHECK_IN_SELECT_MEMBER = 'HealCheckInSelectMember';
export const HEAL_CHECK_IN_FORM = 'HealCheckInForm';
export const HEAL_WALK_IN_SELECT_DOCTOR = 'HealWalkInSelectDoctor';
export const HEAL_WALK_IN_CHECK_IN = 'HealWalkInCheckIn';
export const HEAL_REMOTE_TICKET_BOOKING = 'HealRemoteTicketBooking';
export const HEAL_REMOTE_TICKET_BOOKING_RESULT =
  'HealRemoteTicketBookingResult';
export const HEAL_CHECK_IN_CONFIRMATION = 'HealCheckInConfirmation';
export const HEAL_CHECK_IN_SCREEN = 'HealCheckInScreen';
export const HEAL_DELETE_QUEUE_CONFIRMATION = 'HealDeleteQueueConfirmation';
export const HEAL_CONFIRM_NEXT = 'HealConfirmNext';
export const HEAL_CLINIC_INFORMATION = 'HealClinicInformation';
