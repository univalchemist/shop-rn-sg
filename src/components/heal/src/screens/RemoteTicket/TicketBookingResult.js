import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import {
  SafeAreaView,
  Dimensions,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  Image,
} from 'react-native';
import PropTypes from 'prop-types';

import { useIntl, useTheme } from '@heal/src/wrappers/core/hooks';
import {
  Box,
  PlainText,
  TrackedCarousel,
  Flex,
  Text,
  SectionHeadingText,
  Divider,
} from '@heal/src/wrappers/components';
import { ClinicInfo } from '@heal/src/components';

import { successImage } from '@heal/images';
import { HEAL_APPOINTMENT_LIST } from '@heal/routes';
import { DOCTOR_LANDING } from '@routes';
import theme from '@theme';

const styles = StyleSheet.create({
  scrollViewContainer: {
    paddingHorizontal: 32,
    paddingVertical: 32,
  },
  successImage: {
    alignSelf: 'center',
    marginTop: 26,
  },
  clinicInfo: {
    marginTop: 24,
    marginHorizontal: 16,
  },
  appointmentButton: {
    height: 48,
    marginVertical: 16,
    marginHorizontal: 32,
  },
});

const TicketBookingResult = ({
  route,
  navigation,
  bookedData,
  detailsClinic,
}) => {
  const theme = useTheme();
  const intl = useIntl();
  const { doctorIndex } = route.params;

  return (
    <Box flex={1} as={SafeAreaView}>
      <ScrollView contentContainerStyle={styles.scrollViewContainer}>
        <Text
          fontSize={32}
          fontWeight={300}
          lineHeight={37}
          letterSpacing={-1.5}
          textAlign="center"
          color={theme.heal.colors.gray[3]}
        >
          {intl.formatMessage({ id: 'heal.ticketBookingResult.thankYou' })}
        </Text>
        <Image source={successImage} style={styles.successImage} />
        <ClinicInfo
          style={styles.clinicInfo}
          doctorName={detailsClinic.doctors[doctorIndex].name}
          address={detailsClinic.address}
          date={detailsClinic.date}
        />
      </ScrollView>
      <TouchableOpacity
        onPress={() => {
          navigation.navigate(DOCTOR_LANDING, { tab: 1 });
        }}
      >
        <Divider full />
        <Box
          style={styles.appointmentButton}
          backgroundColor={theme.colors.red}
          alignItems="center"
          justifyContent="center"
        >
          <PlainText color={theme.colors.white}>
            {intl.formatMessage({
              id: 'heal.ticketBookingResult.viewButton',
            })}
          </PlainText>
        </Box>
      </TouchableOpacity>
    </Box>
  );
};

const mapStateToProps = ({
  heal: {
    detailsClinic,
    remoteTicket: { bookedData },
  },
}) => ({ detailsClinic, bookedData });

export default connect(mapStateToProps, {})(TicketBookingResult);
