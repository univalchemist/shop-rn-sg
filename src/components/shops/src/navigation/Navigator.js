/* istanbul ignore file */
//TODO update unit test
import React from 'react';
import {
  createStackNavigator,
  TransitionPresets,
} from '@react-navigation/stack';
import { connect } from 'react-redux';
import { useIntl } from '@shops/wrappers/core/hooks';
import {
  HomeScreen,
  SelectCategoryScreen,
  FilterByScreen,
  SortByScreen,
  ProductDetailsScreen,
  ReviewScreen,
} from '@shops/screens/home';
import {
  AddAddressScreen,
  DeliveryAddressesScreen,
  EditAddressScreen,
} from '@shops/screens/deliveryAddress';
import { ModalBackButton } from '@shops/wrappers/components';
import { ShopHeader, ClearAllButton } from '@shops/components';
import {
  CartScreen,
  EditProductScreen,
  CheckoutScreen,
  PaymentScreen,
  PaymentSuccessScreen,
  PaymentFailedScreen,
  PaymentPendingScreen,
} from '@shops/screens/cart';
import { updateSortType, updateFilterTypes } from '@shops/store/actions';
import { StackBackButton } from '@shops/wrappers/components';
import * as routes from './routes';
import {
  OrderHistoryDetailScreen,
  OrderHistoryScreen,
  ReviewSubmittedScreen,
  TrackOrderScreen,
  WriteReviewScreen,
} from '@shops/screens/orderhistory';
import { SingleSelectModal } from '@wrappers/components/form';
import LogScreen from '@shops/screens/log/LogScreen';
import SelectAddressModal from '@shops/screens/home/ProductDetailsScreen/SelectAddressModal';

const Stack = createStackNavigator();

const Navigator = ({ updateSortType, updateFilterTypes }) => {
  const intl = useIntl();

  const onClearAllSortByPress = () => {
    updateSortType(null);
  };

  const onClearAllFilterByPress = () => {
    updateFilterTypes({});
  };

  return (
    <Stack.Navigator
      initialRouteName={routes.NAVIGATION_HOME_SCREEN_PATH}
      screenOptions={{ headerBackTitleVisible: false }}
      headerMode="float"
    >
      <Stack.Screen
        name={routes.NAVIGATION_HOME_SCREEN_PATH}
        component={HomeScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.NAVIGATION_HOME_PRODUCT_PATH}
        component={ProductDetailsScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CART_SCREEN}
        component={CartScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CHECKOUT_SCREEN}
        component={CheckoutScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_PRODUCT_SCREEN}
        component={EditProductScreen}
        options={({ route, navigation }) => {
          const onUpdateFinish = route?.params?.onUpdateFinish;
          return {
            title: intl.formatMessage({ id: 'shop.screenTitle.editProduct' }),
            headerLeft: () => (
              <StackBackButton
                icName={'close'}
                onPress={() => {
                  navigation.goBack();
                  onUpdateFinish?.();
                }}
              />
            ),
            ...TransitionPresets.ModalTransition,
          };
        }}
      />
      <Stack.Screen
        name={routes.SHOP_SELECT_CATEGORY_MODAL}
        component={SelectCategoryScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.selectCategory' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_SORT_MODAL}
        component={SortByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.sortBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllSortByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_FILTER_MODAL}
        component={FilterByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.filterBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllFilterByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />

      <Stack.Screen
        name={routes.SELECT_ADDRESS_MODAL}
        component={SelectAddressModal}
        options={() => ({
          title: intl.formatMessage({ id: 'shop.product.selectionLocation' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_REVIEW_SCREEN}
        component={ReviewScreen}
        options={({ route }) => {
          const ratingCount = route.params?.reviews?.length || 0;
          const title = intl.formatMessage(
            {
              id: 'shop.review.title',
              defaultMessage: `Review(${ratingCount})`,
            },
            { ratingCount },
          );

          return {
            title,
            headerBackImage: () => <ModalBackButton />,
            ...TransitionPresets.ModalTransition,
          };
        }}
      />
      <Stack.Screen
        name={routes.SHOP_DELIVERY_ADDRESSES_SCREEN}
        component={DeliveryAddressesScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_ADD_ADDRESS_SCREEN}
        component={AddAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.addNewAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_ADDRESS_SCREEN}
        component={EditAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.editAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.ORDER_HISTORY_SCREEN}
        component={OrderHistoryScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={OrderHistoryDetailScreen}
        name={routes.ORDER_HISTORY_DETAIL_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={TrackOrderScreen}
        name={routes.TRACK_ORDER_SCREEN}
        options={{
          title: intl.formatMessage({
            id: 'shop.orderHistory.trackOrderTitle',
          }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        component={WriteReviewScreen}
        name={routes.WRITE_REVIEW_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={ReviewSubmittedScreen}
        name={routes.REVIEW_SUBMITTED_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={PaymentScreen}
        name={routes.PAYMENT_SCREEN}
        options={({ navigation }) => ({
          headerLeft: () => (
            <StackBackButton
              onPress={() => {
                navigation.navigate(routes.SHOP_CART_SCREEN);
              }}
            />
          ),
          title: intl.formatMessage({
            id: 'shop.payment.title',
          }),
        })}
      />
      <Stack.Screen
        component={PaymentSuccessScreen}
        name={routes.PAYMENT_SUCCESS_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={PaymentFailedScreen}
        name={routes.PAYMENT_FAILED_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={PaymentPendingScreen}
        name={routes.PAYMENT_PENDING_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        component={LogScreen}
        name={'LogScreen'}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_SINGLE_SELECT_MODAL}
        component={SingleSelectModal}
        options={({ route }) => ({
          title: route.params?.title,
          headerBackImage: ModalBackButton,
          ...TransitionPresets.ModalTransition,
        })}
      />
    </Stack.Navigator>
  );
};

export const ShopNavigator = connect(null, {
  updateSortType,
  updateFilterTypes,
})(Navigator);
