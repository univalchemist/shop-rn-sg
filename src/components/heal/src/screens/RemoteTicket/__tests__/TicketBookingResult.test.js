import React from 'react';
import { TouchableOpacity } from 'react-native';
import { render, renderForTest } from '@testUtils';
import navigation from '@testUtils/__mocks__/navigation';
import { flushMicrotasksQueue, fireEvent } from 'react-native-testing-library';

import TicketBookingResult from '../TicketBookingResult';
import { DOCTOR_LANDING } from '@routes';
import { HEAL_APPOINTMENT_LIST } from '@heal/routes';

const detailsClinic = {
  id: 532,
  name: 'Clinic Icity',
  latitude: 3.1267439,
  longitude: 101.6424801,
  qrCode: '2502',
  district: 'Klang',
  distanceToClient: 11300173.986425,
  address: '2502, JALAN ICITY',
  openingHours:
    'Sun: 00:00-00:00, 00:00-00:00\nMon: 08:00-13:00, 14:00-23:00\nTue: 08:00-13:00, 14:00-23:00\nWed: 08:00-13:00, 14:00-23:00\nThu: 08:00-13:00, 14:00-23:00\nFri: 08:00-13:00, 14:00-23:00\nSat: 00:00-00:00, 08:00-13:00\nPublic Holiday: 00:00-00:00, 00:00-00:00\n',
  clinicProviderId: 1,
  isActive: true,
  phoneNumber: '0703610700',
  doctors: [
    {
      id: 322,
      locale: 'en-HK',
      name: 'Yeo An Shin',
      gender: 'Male',
      isActive: true,
      clinics: [],
      clinicProviderId: 1,
    },
    {
      id: 323,
      locale: 'en-HK',
      name: 'Clinic Icity',
      gender: 'Male',
      isActive: true,
      clinics: [],
      clinicProviderId: 1,
    },
  ],
};

const bookedData = {
  value: {
    id: 27,
    status: 'Requested',
    token: '304|1590597845|602',
    estimatedConsultationTime: '12:44 am',
    position: 0,
    type: 'RemoteTicket',
    memberId: 392,
    clinicProviderId: 1,
    doctorId: 694,
    clinicId: 748,
  },
};

describe('TicketBookingResult', () => {
  test('Render TicketBookingResult', async () => {
    const component = renderForTest(
      <TicketBookingResult
        navigation={navigation}
        route={{ params: { doctorIndex: 0 } }}
      />,
      {
        initialState: {
          heal: {
            detailsClinic,
            remoteTicket: { bookedData },
          },
        },
      },
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should press view appointment button', async () => {
    const [screen] = render(
      <TicketBookingResult
        navigation={navigation}
        route={{ params: { doctorIndex: 0 } }}
      />,
      {
        initialState: {
          heal: {
            detailsClinic,
            remoteTicket: { bookedData },
          },
        },
      },
    );

    await flushMicrotasksQueue();

    const items = screen.queryAllByType(TouchableOpacity);
    fireEvent.press(items[0]);
    expect(navigation.navigate).toBeCalledWith(DOCTOR_LANDING, { tab: 1 });
  });
});
