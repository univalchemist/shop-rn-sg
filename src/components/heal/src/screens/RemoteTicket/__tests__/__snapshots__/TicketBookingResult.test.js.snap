// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketBookingResult Render TicketBookingResult 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        flex={1}
                        style={
                          Array [
                            Object {
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          contentContainerStyle={
                            Object {
                              "paddingHorizontal": 32,
                              "paddingVertical": 32,
                            }
                          }
                        >
                          <View>
                            <Text
                              color="#212121"
                              fontFamily="default"
                              fontSize={32}
                              fontWeight={300}
                              letterSpacing={-1.5}
                              lineHeight={37}
                              style={
                                Array [
                                  Object {
                                    "color": "#212121",
                                    "fontFamily": "Arial",
                                    "fontSize": 32,
                                    "fontWeight": "300",
                                    "letterSpacing": -1.5,
                                    "lineHeight": 37,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                              textAlign="center"
                            >
                              Thank you for booking with us!
                            </Text>
                            <Image
                              source={
                                Object {
                                  "testUri": "../../../src/components/heal/images/successImage.png",
                                }
                              }
                              style={
                                Object {
                                  "alignSelf": "center",
                                  "marginTop": 26,
                                }
                              }
                            />
                            <View
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "marginHorizontal": 16,
                                    "marginTop": 24,
                                  },
                                ]
                              }
                            >
                              <View
                                flexDirection="row"
                                mb={25}
                                pt={20}
                                style={
                                  Array [
                                    Object {
                                      "flexDirection": "row",
                                      "marginBottom": 25,
                                      "paddingTop": 20,
                                    },
                                  ]
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "testUri": "../../../src/components/heal/images/appointmentCalender.png",
                                    }
                                  }
                                  style={
                                    Array [
                                      Object {},
                                      Object {
                                        "tintColor": "#212121",
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    Array [
                                      Object {},
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    letterSpacing={0.3}
                                    lineHeight={22}
                                    ml={18}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "letterSpacing": 0.3,
                                          "lineHeight": 22,
                                          "marginLeft": 18,
                                        },
                                        Object {
                                          "color": "#727272",
                                          "fontSize": 14,
                                          "letterSpacing": 0.25,
                                          "lineHeight": 20,
                                        },
                                      ]
                                    }
                                  >
                                    11 Jun 2020
                                  </Text>
                                </View>
                              </View>
                              <View
                                alignItems="center"
                                flexDirection="row"
                                mr="auto"
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "marginRight": "auto",
                                    },
                                  ]
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "testUri": "../../../src/components/heal/images/icon-doctor.png",
                                    }
                                  }
                                  style={
                                    Array [
                                      Object {},
                                    ]
                                  }
                                />
                                <View
                                  ml={18}
                                  style={
                                    Array [
                                      Object {
                                        "marginLeft": 18,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    letterSpacing={0.3}
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "letterSpacing": 0.3,
                                          "lineHeight": 22,
                                        },
                                        Object {
                                          "color": "#727272",
                                          "fontSize": 14,
                                          "letterSpacing": 0.25,
                                          "lineHeight": 20,
                                        },
                                      ]
                                    }
                                  >
                                    Dr. Yeo An Shin
                                  </Text>
                                </View>
                              </View>
                              <View
                                alignItems="center"
                                flexDirection="row"
                                mr="auto"
                                mt={24}
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "marginRight": "auto",
                                      "marginTop": 24,
                                    },
                                  ]
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "testUri": "../../../src/components/heal/images/location.png",
                                    }
                                  }
                                  style={
                                    Array [
                                      Object {},
                                    ]
                                  }
                                />
                                <Text
                                  color="gray.1"
                                  fontFamily="default"
                                  fontSize={16}
                                  letterSpacing={0.3}
                                  lineHeight={22}
                                  ml={18}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#646464",
                                        "fontFamily": "Arial",
                                        "fontSize": 16,
                                        "letterSpacing": 0.3,
                                        "lineHeight": 22,
                                        "marginLeft": 18,
                                      },
                                      Object {
                                        "color": "#727272",
                                        "fontSize": 14,
                                        "letterSpacing": 0.25,
                                        "lineHeight": 20,
                                      },
                                    ]
                                  }
                                >
                                  2502, JALAN ICITY
                                </Text>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                        <View
                          accessible={true}
                          focusable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            backgroundColor="gray.5"
                            height={1}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#e3e3e3",
                                  "height": 1,
                                  "width": "100%",
                                },
                              ]
                            }
                            width="100%"
                          />
                          <View
                            alignItems="center"
                            backgroundColor={
                              Array [
                                "#DB0011",
                              ]
                            }
                            justifyContent="center"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#DB0011",
                                  "justifyContent": "center",
                                },
                                Object {
                                  "height": 48,
                                  "marginHorizontal": 32,
                                  "marginVertical": 16,
                                },
                              ]
                            }
                          >
                            <Text
                              color="#ffffff"
                              fontFamily="default"
                              fontSize={16}
                              letterSpacing={0.3}
                              lineHeight={22}
                              style={
                                Array [
                                  Object {
                                    "color": "#ffffff",
                                    "fontFamily": "Arial",
                                    "fontSize": 16,
                                    "letterSpacing": 0.3,
                                    "lineHeight": 22,
                                  },
                                ]
                              }
                            >
                              View my appointments
                            </Text>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
